### DISCUSSION TITLE
MYSQL Tables not showing

### DISCUSSION THREAD

**Discussion by alanfallow** [Budibase/budibase#11504] [Help] [answered]
Good afternoon folks.
I have managed to create a MYSQL data source and connect.  It says fetching tables OK.  But no tables are showing.  Am I missing something.
Is there an easier option than MYSQL.

**Comment by ConorWebb96** [comment]
Hey @alanfallow,

Can you screenshot of your data source connection settings? 
<img width="646" alt="Screenshot 2023-08-11 at 13 44 18" src="https://github.com/Budibase/budibase/assets/126772285/41d13665-cafe-4915-ae18-21175c0369fb">

Are you definitely sure there are tables already existing in there that you should be able to fetch? When you click the fetch tables button, is it completely blank? (It shouldn't say fetching tables unless you have selected your tables and then fetched.)
<img width="646" alt="Screenshot 2023-08-11 at 13 45 16" src="https://github.com/Budibase/budibase/assets/126772285/71fe4d81-ef7e-42e8-9d0d-6d6b25d5b5be">

Make sure you have properly added your primary keys otherwise the tables won't be synced across to Budibase.

To answer your last question, the easiest method would be trying out the internal table as it requires zero configuration outside of Budibase. You can do this by clicking the `Add source` button and then selecting the first option.

<img width="955" alt="Screenshot 2023-08-11 at 13 50 08" src="https://github.com/Budibase/budibase/assets/126772285/d8e606a0-49ab-4c0a-bc6b-76b9842d1507">

**Comment by alanfallow** [comment]
Hi yes it says fetching tables but screen is blank.  I have direct admin which I’m not sure about will try another on my cPanel and als have a go with built in table.  Thanks again will report back.

**Comment by alanfallow** [comment]
Seems to be ok.  I can create tables and rows fine through Budibase now on MySQL.  Thanks again.

**Comment by jbueter** [comment]
@ConorWebb96 I had the same issue, but with a PostgreSQL database. I now read in your comment that a primary key is required to be able to fetch the tables. It would be very helpful if either an error message in Budibase was returned, indicating that no primary key exists, or if you would at least add that information to [your docs](https://docs.budibase.com/docs/postgresql). Thank you anyway for your comment and explanation.

**Reply by ConorWebb96** [reply, marked as answer]
Yes you're right. I think the best route in this case is to throw a descriptive error when there is no primary keys letting the user know what the issue is.

Heres the issue I created for this; https://github.com/Budibase/budibase/issues/11843

I hope this helps.
